packages/moonstone/DatePicker/DatePicker.js

```diff
@@ -37,8 +37,7 @@ const dateTimeConfig = {
 			values.day = value.getDays();
 			values.maxMonths = i18n.formatter.cal.getNumMonths(values.year);
 			values.maxDays = i18n.formatter.cal.getMonLength(values.month, values.year);
-			values.maxYear = i18n.toLocalYear(props.maxYear);
```

Comment:
What do you think of plucking `defaultProps` from `DatePickerBase` here?

```
values.maxYear = i18n.toLocalYear(props.maxYear || DatePickerBase.defaultProps.maxYear);
values.minYear = i18n.toLocalYear(props.minYear || DatePickerBase.defaultProps.minYear);
```

I'm trying to decide which is the right overlap of concerns. Does it make more sense to:
* inject the default values (either statically or via ref to `defaultProps`) in this config object
* delegate i18n formatting down to the component.

Injection duplicates default prop handling from `DatePickerBase` in the common case of i18n support but leaves it with the base component in prerendering. Using `defaultProps` mitigates that overlap a bit by avoiding using different values.

Delegation shifts _one_ piece of i18n formatting to the base component while keeping the rest in `DateTimeDecorator`. I worry that mixed responsibilities creates unclear APIs and can lead to bugs with multiple layers implementing different parts of the same capability. It does, however, keep prop management more clean between the layers.

Reply:
I think your first suggestion is okay in a way that it simplifies the concern. The only reason I didn't follow this way was that I imagine `DateTimeDecorator` (and most of our `*able` HOCs) to be a "behavior-setter" which does not associate with default values.

That being said, I also agree that the latter approach is mixing responsibilities. `DateTimeDecorator` handles i18n adjustments, and `DateTimePickerBase` is a simple render component. And it seems like a "formatter"  lies in between. I'd thought making it an opt-in function would mitigate somewhat, but I don't think that's true.

Reply:
I tend to agree with your definition of our `*able` HOCs. In other cases (`disabled` being an example), we have set a default for props that are required to be valued by the HOC in order for it to operate. `DateTimeDecorator` is an anomaly in its design in that it supports author-land functions that operate on its props. Because of that, I think it's okay to do what is proposed here.

Another option that isn't feasible right now would be to support hoisting statics (`defaultProps` in this case) up the HOC stack to ensure that the right defaults are set on the generated component. That could introduce other issues if we start to assume values have been set due to hoisting rather than explicitly handling them.